### PR TITLE
Additional sBTC clarity mini tests

### PR DESCRIPTION
**Pull request by setzeus**
# sBTC Clarity-Bitcoin-Mini Tests
## Summary of Changes

This PR is meant to add a handful of Clarity unit tests for the clarity-bitcoin-mini.clar file found in romeo/asset-contract. These tests are unit tests for varying clarity & bitcoin logic such as: verify-block-header & verify-merkle-proof.

### Risks

The largest risk with this PR & logic likely lies in the 'prepare' setup & in generating false unit-test positives due to complexity in clarity -> typescripts tests & to required preparation.

### How were these changes tested?

These tests were locally tested using clarinet unit testing, specifically generating the non-traditional ./scripts/test.sh  tests written primarily in Clarity.

To replicate the tests found here, cd into romeo/asset-contract then run ./scripts/test.sh. 

### What future testing should occur?

While this first round of tests focused on the functions mentioned above, it'd make sense to focus future testing on the (was-txid-mined ...) logic as well as additional unit tests on fundamental helper functions such as (reverse-buffs ...).

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by AshtonStephens** [comment]
I think this is the first use of the testing standards doc template and this is a perfect example of how to use it! 10/10 this PR is awesome.

**Review by jo-tm — APPROVED** [review]
LGTM

File: romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        (unwrap-panic (add-burnchain-block-header-hash u2431087 0x0000a02065bc9201b5b5a1d695a18e4d5efe5d52d8ccc4129a2499141d000000000000009160ba7ae5f29f9632dc0cd89f466ee64e2dddfde737a40808ddc147cd82406f18b8486488a127199842cec7))
+        (ok true)))
+
+;; @name check verify-block-header
+(define-public (test-verify-block-header)
+    (let (
+        (burnchain-block-height u807525)
+        ;; block id: 0000000000000000000104cf6179ece70fe28f1f6a24126e8d8c91d42d5eafb4
+        (raw-block-header 0x00001733539613a8d931b08f0d3f746879572a6d3e12623b16d20000000000000000000034f521522fba52c2c5c75609d261b490ee620661319ab23c68f24d756ff4ced801230265ae32051718d9aadb))
+        ;; prepare
```
Comment (friedger) [inline romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar:21]: why is this `prepare`?

File: romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        (hash-wtx-le 0x04117dc370c45b8a44bf86a3ae4fa8d0b186b5b27d50939cda7501723fa12ec6)
+        (hash-wtx-be 0xc62ea13f720175da9c93507db2b586b1d0a84faea386bf448a5bc470c37d1104)
+        (merkle-root 0x15424423c2614c23aceec8d732b5330c21ff3a306f52243fbeef47a192c65c86)
+        (proof {
+            tx-index: u3,
+            hashes: (list 0xb2d7ec769ce60ebc0c8fb9cc37f0ad7481690fc176b82c8d17d3c05da80fea6b 0x122f3217765b6e8f3163f6725d4aa3d303e4ffe4b99a5e85fb4ff91a026c17a8),
+            tree-depth: u2})
+        )
+
+        (ok (is-err (contract-call? .clarity-bitcoin-mini verify-merkle-proof
```
Comment (friedger) [inline romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar:97]: Which error do we expect?

File: romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        ;; hash256 wtx raw hex of tx 0ba1d831915fb5d9bbd4c19665ebf12157ee6aaf1d81e054b667bb5539e77bd8
+        (hash-wtx-le 0x073782910dca06f17d09828a116b1b991e1255d1da2f541a465e681e11ebf32b)
+        ;; witness merkle root found in coinbase op_return (below concatenated with 32-bytes of 0x00 then hash256)
+        (merkle-root 0x73cbac43da0bffeb8161d9da16f9106ea0d13f6c29b4fb031a61282d1b7f4255)
+        (proof {
+            tx-index: u2,
+            hashes: (list 0x1bd75b05fec22e8436fa721115f5a8843ba3d2d17640df9653303b186c6a3701 0x073782910dca06f17d09828a116b1b991e1255d1da2f541a465e681e11ebf32b),
+            tree-depth: u2}))
+
+            (ok (contract-call? .clarity-bitcoin-mini verify-merkle-proof
```
Comment (friedger) [inline romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar:58]: Do we expect `(ok (ok true))` here?

File: romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar (hunk 2 above)
Comment (friedger) [reply romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar:97]: Do we have a test for proof too short?

File: romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar [hunk 4]
```diff
@@ -0,0 +1,103 @@
+(define-constant test-contract-principal (as-contract tx-sender))
+(define-constant zero-address 'SP000000000000000000002Q6VF78)
```
Comment (friedger) [inline romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar:2]: Why do we need this constant?

File: romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar (hunk 3 above)
Comment (friedger) [reply romeo/asset-contract/tests/clarity-bitcoin-mini_test.clar:58]: Is the result always `true`?

**Review by friedger — CHANGES_REQUESTED** [review]
While the test passes, I have some nit.

Also I think it might be good to have the proofs in a list, so that it is very easy to add more. Or to replace an existing one with one that I would like to test from my development environment.

For the failed verification, it would be good to be more explicit why they fail.

**Comment by friedger** [comment]
@MarvinJanssen mentioned some helper functions for asserts. I didn't find them in https://github.com/search?q=repo%3ATrust-Machines%2Fstacks-sbtc+asserts&type=code

**Comment by setzeus** [comment]
@friedger think he meant like the (assert-eq ...) & (assert-all-eq ...) found here: https://github.com/Trust-Machines/stacks-sbtc/blob/main/sbtc-mini/tests/sbtc-registry_test.clar

**Review by friedger — APPROVED** [review]
Looks good

**Comment by friedger** [comment]
@setzeus It would be nice to use these asserts-eq functions

**Review by jo-tm — APPROVED** [review]
LGTM

**Pull request merged**